packages/components/src/internal/components/search/EntityFieldFilterModal.tsx

```diff
@@ -170,6 +198,41 @@ export const EntityFieldFilterModal: FC<Props> = memo(props => {
         [dataTypeFilters, activeQuery]
     );
 
+    const onHasNoValueInQueryChange = useCallback(
+        (check: boolean) => {
+            if (!entityDataType.supportHasNoValueInQuery) return;
+
+            setCardDirty?.(true);
+            setFilterError(undefined);
+            const schemaQuery = entityDataType.getInstanceSchemaQuery(activeQuery);
+            const selectQueryFilterKey = assaySampleIdCols[activeQuery]?.lookupFieldKey;
+            const targetQueryFilterKey = assaySampleIdCols[activeQuery]?.fieldKey;
+            setDataTypeFilters(
+                getDataTypeFiltersWithNotInQueryUpdate(
+                    dataTypeFilters,
+                    schemaQuery,
+                    activeQuery,
+                    selectQueryFilterKey,
+                    targetQueryFilterKey,
+                    check
+                )
+            );
+        },
+        [dataTypeFilters, activeQuery, entityDataType]
+    );
+
+    const hasNotInQueryFilter = useMemo((): boolean => {
+        const activeQueryFilters: FieldFilter[] = dataTypeFilters[activeQuery];
+        if (!activeQueryFilters || activeQueryFilters.length === 0) return false;
+
+        let hasFilter = false;
+        activeQueryFilters.forEach(fieldFilter => {
+            if (fieldFilter.filter.getFilterType().getURLSuffix() === COLUMN_NOT_IN_FILTER_TYPE.getURLSuffix())
+                hasFilter = true;
+        });
+        return hasFilter;
```

Comment:
Could you use `.some()` here to short-circuit?
```suggestion
        return activeQueryFilters.some(fieldFilter =>
            fieldFilter.filter.getFilterType().getURLSuffix() === COLUMN_NOT_IN_FILTER_TYPE.getURLSuffix()
        );
```